File: src/main/java/pl/training/workshop/m6/s01_strategy/step1/PriceBoard.java
package pl.training.workshop.m6.s01_strategy.step1;

import pl.training.workshop.m6.s01_strategy.PriceRequest;
import pl.training.workshop.shared.Money;

/** Krok 1: klient bez zmian. */
public final class PriceBoard {
    private final TicketPricer pricer = new TicketPricer();

    public Money priceFor(PriceRequest request) {
        return pricer.price(request.base(), request.ticketType(), request.program());
    }
}

File: typescript/src/workshop/m6/s01_strategy/start/PriceBoard.ts
import type { Money } from '../../../shared/Money.js';
import type { PriceRequest } from '../PriceRequest.js';
import { TicketPricer } from './TicketPricer.js';

/** Start: klient cennika - kasa wycenia bilet wg programu z konfiguracji. */
export class PriceBoard {
  private readonly pricer = new TicketPricer();

  priceFor(request: PriceRequest): Money {
    return this.pricer.price(request.base, request.ticketType, request.program);
  }
}
